<?xml version="1.0" ?>
<robot xmlns:xacro="http://www.ros.org/wiki/xacro" name="two_link">

<xacro:property name="filename" value="$(find mnist_digit_tracker)/config/mnist_digit_tracker.yaml"/>
<xacro:property name="props" value="${load_yaml(filename)}"/>
<xacro:property name="l1" value="${props['tracker']['two_link']['l1']}"/>
<xacro:property name="l2" value="${props['tracker']['two_link']['l2']}"/>

  <material name="black">
    <color rgba="0.0 0.0 0.0 1.0"/>
  </material>
  <material name="orange">
    <color rgba="1.0 0.4 0.0 1.0"/>
  </material>

  <link name="base"/>

  <joint name="fixed" type="fixed">
    <parent link="base"/>
    <child link="link1"/>
  </joint>

  <link name="link1">
    <collision>
      <origin xyz="0 0 0.05" rpy="0 0 0"/>
      <geometry>
        <box size="0.1 0.1 0.1"/>
      </geometry>
    </collision>
    <visual>
      <origin xyz="0 0 0.05" rpy="0 0 0"/>
      <geometry>
        <box size="0.1 0.1 0.1"/>
      </geometry>
      <material name="black"/>
    </visual>
    <inertial>
      <origin xyz="0 0 0.05" rpy="0 0 0"/>
      <mass value="1"/>
      <inertia ixx="1.0" ixy="0.0" ixz="0.0" iyy="1.0" iyz="0.0" izz="1.0"/>
    </inertial>
  </link>

  <joint name="joint1" type="revolute">
    <parent link="link1"/>
    <child link="link2"/>
    <origin xyz="0 0 0.1" rpy="0 0 -1.57"/>
    <axis xyz="0 0 1"/>
    <limit effort="30" lower="-3.14" upper="3.14" velocity="3.14"/>
  </joint>

  <link name="link2">
    <collision>
      <origin xyz="0 ${l1/2} 0" rpy="-1.57 0 0"/>
      <geometry>
        <box size="0.1 0.1 ${l1}"/>
      </geometry>
    </collision>
    <visual>
      <origin xyz="0 ${l1/2} 0" rpy="-1.57 0 0"/>
      <geometry>
        <box size="0.1 0.1 ${l1}"/>
      </geometry>
      <material name="orange"/>
    </visual>
    <inertial>
      <origin xyz="0 ${l1/2} 0" rpy="-1.57 0 0"/>
      <mass value="1"/>
      <inertia ixx="1.0" ixy="0.0" ixz="0.0" iyy="1.0" iyz="0.0" izz="1.0"/>
    </inertial>
  </link>

  <joint name="joint2" type="revolute">
    <parent link="link2"/>
    <child link="link3"/>
    <origin xyz="0 ${l1} 0" rpy="0 0 0"/>
    <axis xyz="0 0 1"/>
    <limit effort="30" lower="-3.14" upper="3.14" velocity="3.14"/>
  </joint>

  <link name="link3">
    <collision>
      <origin xyz="0 ${l1/2} 0" rpy="-1.57 0 0"/>
      <geometry>
        <box size="0.1 0.1 ${l1}"/>
      </geometry>
    </collision>
    <visual>
      <origin xyz="0 ${l1/2} 0" rpy="-1.57 0 0"/>
      <geometry>
        <box size="0.1 0.1 ${l1}"/>
      </geometry>
      <material name="black"/>
    </visual>
    <inertial>
      <origin xyz="0 ${l1/2} 0" rpy="-1.57 0 0"/>
      <mass value="1"/>
      <inertia ixx="1.0" ixy="0.0" ixz="0.0" iyy="1.0" iyz="0.0" izz="1.0"/>
    </inertial>
  </link>

</robot>
